### ISSUE TITLE
Support to doctrine/mongodb-odm-bundle 4.x still with errors

### ISSUE THREAD

**Issue by lcavero** [markitosgv/JWTRefreshTokenBundle#169]
Hi, lookslike the new version 0.8 doesn't fix all errors.
Now, when I require the library I get that:

The mapping file /xxx/vendor/gesdinet/jwt-refresh-token-bundle/Resources/config/mongodb/doctrine-document/RefreshToken.mongodb.xml is invalid:                                                          
!!    Line 8:0: Element '{http://doctrine-project.org/schemas/odm/doctrine-mongo-  
!!    mapping}document', attribute 'repositoryClass': The attribute 'repositoryClass' is not allowed.                                                         
!!                                                                                 
!!    Line 9:0: Element '{http://doctrine-project.org/schemas/odm/doctrine-mongo-  
!!    mapping}field', attribute 'fieldName': The attribute 'fieldName' is not allowed.   

Thanks for updating the library so fast!

**Comment by markitosgv** [comment]
@lcavero could you please test against dev-master , I fix xml mapping

**Comment by pandresp** [comment]
Hi,
I really need this fix.
When we can expect this amendment to be on the master?

**Comment by markitosgv** [comment]
@pandresp could you please update to dev-master to test if this fix is working?

**Comment by pandresp** [comment]
@markitosgv on `dev-master` the previous error is not ocured,
but new one exists I think.

Is the table name was changed?
`Base table or view not found: 1146 Table 'refresh_token' doesn't exist"`
Do you now something about this?

In data base I have table named as `refresh_tokens` with `s` on end.
On version 0.7.1 working with `refresh_tokens` table.

**Comment by markitosgv** [comment]
Yes, but look at this:

https://github.com/markitosgv/JWTRefreshTokenBundle/blob/40f03d12d8167f5fd1124ef9befb8ac3b07e1f87/Resources/config/mongodb/doctrine-document/RefreshToken.mongodb.xml#L8

Are you checking against last dev-master?

**Comment by lcavero** [comment]
@markitosgv Hi, testing in dev-master solves the previous problem, a new problem happens now

  No identifier/primary key specified for Document 'Gesdinet\JWTRefreshTokenB  
!!    undle\Document\RefreshToken'. Every Document must have an identifier/primar  
!!    y key.     
Ty

**Comment by markitosgv** [comment]
Ok @lcavero 

Could you please test chang line 9 to this:

```
<field field-name="id" type="id" id="true" />
```

**Comment by pandresp** [comment]
@markitosgv, I think you do not understand exactly my case

In my project I use postgres DB for entity (and for refresh tokens).
In last days I ad hybrid configuration postgres + mongoDB
And then problems was started.

For clear, I want use postgres DB to store my entities and refresh tokens.
Only for some dokuments I use mongoDB.

I see that you switch configuration form `yaml` to `xml`
After switch the configuration is not the same
```yaml
Gesdinet\JWTRefreshTokenBundle\Entity\RefreshToken:
  type: entity
  table: refresh_tokens
  repositoryClass: Gesdinet\JWTRefreshTokenBundle\Entity\RefreshTokenRepository
```
```xml
<entity name="Gesdinet\JWTRefreshTokenBundle\Entity\RefreshToken">
```
shouldbe:
```xml
<entity name="Gesdinet\JWTRefreshTokenBundle\Entity\RefreshToken" table="refresh_tokens">
```
I think you forgot the table name

**Comment by markitosgv** [comment]
You're right @pandresp , mistake was introduced in PR #171 . Solved in master and waiting for a new release when @lcavero confirms

**Comment by lcavero** [comment]
@markitosgv 

  The mapping file /home/luilli/FREELANCE/fly/core/vendor/gesdinet/jwt-refresh-token-bundle/Resources/config/mongodb/doctrine-document/RefreshToken.mongodb.xml is invalid:   
  Line 9:0: Element '{http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping}field', attribute 'id': The attribute 'id' is not allowed.

**Comment by markitosgv** [comment]
Ok sorry @lcavero try this:

```
<id field-name="id" />
```

**Comment by lcavero** [comment]
Lookslike it works, no errors in cache clear.
Great!

**Comment by lcavero** [comment]
Yes, it generates the refresh token on login. Perfect!

**Comment by lcavero** [comment]
@markitosgv Hola Marcos, no sé si me he perdido algo pero te cuento. 
Para doctrine odm 4 si no me equivoco tienen que haber dos cosas.
El archivo RefreshToken.mongodb.xml tiene que estar como lo pusiste antes, con el campo <id field-name="id">, pero el archivo AbstractRefreshToken  tiene que estar el tipo "datetime" cambiado a "date". 
He buscado en varias ramas del proyecto y en unas pone una cosa y en otras la otra, pero no las dos a la vez.
Se me ha escapado a mí algo o hay algo mal?
Creo que a lo mejor lo habéis vuelto a poner como datetime, porque yo llegué a probarlo y funcionaba.
Saludos y muchas gracias por actualizar el proyecto

**Comment by shakaran** [comment]
@lcavero, this got sorted out in the thread and then left open. The mapping has been correct since v0.10.0, and it is now checked on every CI run rather than by asking people to try `dev-master`.

Current file:

```xml
<mapped-superclass name="Gesdinet\JWTRefreshTokenBundle\Document\RefreshToken"
                   collection="refresh_tokens"
                   repository-class="Gesdinet\JWTRefreshTokenBundle\Document\RefreshTokenRepository">
    <id />
    <field field-name="refreshToken" type="string" unique="true" />
    <field field-name="username" type="string" />
    <field field-name="valid" type="date"/>
</mapped-superclass>
```

Every attribute the ODM 4.x schema rejected is gone: `repositoryClass` became `repository-class`, `fieldName` became `field-name`, and `<id />` is the identifier — which is the change @markitosgv was walking you towards in the last comment, and it is what fixed the `No identifier/primary key specified for Document` you hit next.

**What has changed since is how it is verified.** The functional tests build a `SimplifiedXmlDriver` over this exact file, and CI runs them against a real MongoDB on every push, so a mapping attribute that a newer ODM schema rejects fails the build. In 2019 this had to be found by installing it.

**@pandresp**, your `refresh_token` versus `refresh_tokens` question was never really answered. The collection here is `refresh_tokens`, plural, and always has been. If a `refresh_token` singular appeared for you, it will have come from a document class of your own without an explicit collection name, where Doctrine derives one from the class name. Naming it settles it:

```php
#[ODM\Document(collection: 'refresh_tokens')]
class RefreshToken extends BaseRefreshToken
{
}
```

Your other point stands too — you said @markitosgv had not quite understood your case, and reading it back, you were right: you were being pointed at the mapping file while your problem was on the ORM side of a project using both.

This is the same underlying change as #174, from the other direction. Closing both.
